pkg/process/discover.go

```diff
@@ -91,30 +93,55 @@ func (a *Analyzer) findProcessID(target *TargetArgs) (int, error) {
 
 			pid, err := strconv.Atoi(dname)
 			if err != nil {
-				return 0, err
+				log.Logger.V(1).Error(err, "creating pid")
+				return nil
 			}
 
-			exeName, err := os.Readlink(path.Join("/proc", dname, "exe"))
+			exeFullPath, err := os.Readlink(path.Join("/proc", dname, "exe"))
 			if err != nil {
 				// Read link may fail if target process runs not as root
-				cmdLine, err := ioutil.ReadFile(path.Join("/proc", dname, "cmdline"))
+				cmdline, err := os.ReadFile(path.Join("/proc", dname, "cmdline"))
 				if err != nil {
-					return 0, err
+					log.Logger.V(1).Error(err, "reading cmdline")
+					return nil
 				}
+				exeFullPath = string(cmdline)
+			}
 
-				if strings.Contains(string(cmdLine), target.ExePath) {
-					return pid, nil
-				}
-			} else if exeName == target.ExePath {
-				return pid, nil
+			exe := filepath.Base(exeFullPath)
+
+			if _, ok := a.ignoreProcesses[exe]; ok {
+				continue
+			}
+			if target != nil && !strings.Contains(string(exeFullPath), target.ExePath) {
```

Comment:
If I'm running in single-target mode, will this keep looping through processes after the target exe has been found?

Reply:
Yes. Currently, once we have the pid and start monitoring the process. If the process dies, auto-instrumentation doesn't try to find the new pid. It doesn't allow us to run on bare metal or as a deployment.

Reply:
I mean, if I set a targetExe will this loop continue searching all pids even after the target pid is found (unnecessarily)? Would you maybe want to update L123 to something like:

```go
pids[pid] = exe
if strings.Contains(string(exeFullPath), target.ExePath) {
  break
}
```